core/src/main/java/hudson/console/ModelHyperlinkNote.java

```diff
@@ -54,6 +54,13 @@ public static String encodeTo(Node node) {
         return encodeTo("/computer/" + nodePath, node.getDisplayName());
     }
 
+    /**
+     * @since TODO
+     */
+    public static String encodeTo(Label label) {
+        return encodeTo("/" + label.getUrl(), label.getName());
```

Comment:
As in https://github.com/jenkinsci/jenkins/blob/cc27faa48d8c5a0cf421a5d648231b506391c2eb/core/src/main/java/hudson/model/Label.java#L162 the URL is encoded automatically, so it does not matter if the label includes URL metacharacters.